inc/add_site_ui/namespace.php

```diff
@@ -411,11 +457,12 @@ function handle_subdirectory( string $value ) : ?array {
 	$network_url = wp_parse_url( network_site_url() );
 	$network_host = $network_url['host'];
 
-	$path = trim( $value, '/' );
+	// Break out URL query parameters and hashes.
+	$path = trim( preg_split( '/(&+|\?+|#+)/', $value )[0], '/' );
```

Comment:
I'm not sure the pluses are necessary here, and given that it's just matching on single characters you could use a character class instead eg:
```suggestion
	$path = trim( preg_split( '/[&\?#]/', $value )[0], '/' );
```

Reply:
Also the ampersand might not be necessary but doesn't do any harm.